TST: wheel builds are failing on Linux

last known successful run: (2025-03-10)
https://github.com/astropy/astropy/actions/runs/22887215041

first known failed run:
https://github.com/astropy/astropy/actions/runs/22936876056

The problem manifests as a build error in `astropy/stats/_stats.c`

```
2026-03-11T04:34:14.2684397Z astropy/stats/_stats.c: In function ‘__Pyx_Get_FusedFunction_Type’:
2026-03-11T04:34:14.2790053Z astropy/stats/_stats.c:36115:15: error: passing argument 1 of ‘Py_INCREF’ from incompatible pointer type [-Wincompatible-pointer-types]
2026-03-11T04:34:14.2791068Z 36115 |     Py_INCREF(tp);
2026-03-11T04:34:14.2791409Z       |               ^~
2026-03-11T04:34:14.2791708Z       |               |
2026-03-11T04:34:14.2792128Z       |               PyTypeObject * {aka struct _typeobject *}
2026-03-11T04:34:14.2792829Z In file included from /opt/_internal/cpython-3.11.14/include/python3.11/Python.h:44,
2026-03-11T04:34:14.2793504Z                  from astropy/stats/_stats.c:55:
2026-03-11T04:34:14.2795062Z /opt/_internal/cpython-3.11.14/include/python3.11/object.h:491:40: note: expected ‘PyObject *’ {aka ‘struct _object *’} but argument is of type ‘PyTypeObject *’ {aka ‘struct _typeobject *’}
2026-03-11T04:34:14.2796155Z   491 | static inline void Py_INCREF(PyObject *op)
2026-03-11T04:34:14.2796631Z       |                              ~~~~~~~~~~^~
2026-03-11T04:34:14.7250956Z error: command '/opt/rh/gcc-toolset-14/root/usr/bin/gcc' failed with exit code 1
2026-03-11T04:34:15.1055732Z 
2026-03-11T04:34:15.1056313Z ERROR Backend subprocess exited when trying to invoke build_wheel
```

the actual source, `astropy/stats/_stats.pyx`, hasn't be updated for several months, so clearly something else changed in the build env.
I would normally suspect code generation (Cython) first, but the timing is off: their latest release was in february. Instead, the most likely culprit could be numpy 2.4.3, released on 2025-03-09.

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -43,14 +43,7 @@ jobs:
       anaconda_user: astropy
       anaconda_package: astropy
       anaconda_keep_n_latest: 10
-
-      # For nightly wheels as well as when building with the 'Build all wheels' label, we disable
-      # the build isolation and explicitly install the latest developer version of numpy as well as
-      # the latest stable versions of all other build-time dependencies.
-      # KEEP IN SYNC WITH [build-system] from pyproject.toml
       env: |
-        CIBW_BEFORE_BUILD: '${{ ((github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' || contains(github.event.pull_request.labels.*.name, 'Build all wheels')) && 'pip install -U --pre --extra-index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple setuptools>=77.0.0 setuptools_scm cython numpy>=0.0.dev0 extension-helpers pyerfa') || '' }}'
-        CIBW_BUILD_FRONTEND: '${{ ((github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' || contains(github.event.pull_request.labels.*.name, 'Build all wheels')) && 'build; args: --no-isolation') || 'build' }}'
         EXTENSION_HELPERS_PY_LIMITED_API: 'cp311'
 
       test_extras: test

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -176,7 +176,6 @@ volint = "astropy.io.votable.volint:main"
 wcslint = "astropy.wcs.wcslint:main"
 
 [build-system]
-# keep in sync with .github/workflows/publish.yml
 requires = ["setuptools>=77.0.0",
             "setuptools_scm>=8.0.0",
             "cython>=3.0.0, <4",

